### DISCUSSION TITLE
Excluding self from one_to_many

### DISCUSSION THREAD

**Discussion by Wardrop** [jeremyevans/sequel#1976] [General]
This seems like it should be obvious.

Let's say I have a "Cars" table/model and I want have a one_to_many named "other_cars" that gets every other car but excludes the current car record. How does one do that exactly? I normally use the ":conditions" option to filter out unwanted records, but given the current record isn't something I can statically define, I'm not sure how to go about this.

**Comment by jeremyevans** [comment]
You can use the `:dataset` association option for this:

```ruby
class Car < Sequel::Model
  one_to_many :other_cars, class: self, dataset: proc{|r| r.associated_class.exclude(id: id)}
end
```

**Comment by Wardrop** [comment]
Excellent, thanks very much. I was going through all the association doco's and did see that `dataset` option but just wasn't sure what the most appropriate or elegant method was.
